packages/viewer/src/components/viewer/viewer.tsx

```diff
@@ -369,6 +374,32 @@ export class Viewer {
     });
   }
 
+  private reconnectStreamingClient(streamId: UUID.UUID): Promise<string> {
+    this.streamDisposables.forEach(d => d.dispose());
+    this.streamDisposables = [];
+    this.stream = new StreamApi(
+      new WebSocketClient(),
+      this.stream.getUrlProvider()
+    );
+    this.setupStreamListeners();
+
+    return new Promise(async resolve => {
+      try {
+        const dispose = await this.commands.execute<Disposable>(
+          'stream.reconnect',
+          streamId,
+          this.dimensions
+        );
+        this.streamDisposables.push(dispose);
+        resolve(streamId);
+      } catch (e) {
+        this.errorMessage = 'Unable to maintain connection to Vertex';
+        throw new WebsocketConnectionError(this.errorMessage, e);
+      }
+      resolve(streamId);
+    });
```

Comment:
There's quite a bit of duplication here with `connectStreamingClient`. Consider extracting this out into another method that can be shared between `connectStreamingClient` and `reconnectStreamingClient`. You can use a function to parameterize the command execution which is different between these calls. Rough implementation might look like:

```ts
  private connectStream(connection: Promise<Disposable>): Promise<void> {
    this.stream = new StreamApi(
      new WebSocketClient(),
      this.stream.getUrlProvider()
    );

    return new Promise(async () => {
      try {
        const dispose = await connection;
        this.setupStreamListeners();
        this.streamDisposables.push(dispose);
      } catch (e) {
        this.errorMessage = 'Unable to maintain connection to Vertex';
        throw new WebsocketConnectionError(this.errorMessage, e);
      }
    });
  }

  private async connectStreamingClient(resource: string): Promise<string | null | undefined> {
    const scene = SceneResource.fromUrn(resource);
    await this.connectStream(
      this.commands.execute<Disposable>('stream.connect', { sceneId: scene.id })
    );

    const response = await this.stream.startStream({ ...this.dimensions });
    
    // note: this is different from previous implementation, and returns the scene view id
    // vs settings it. think this is better as it keeps the implementation more like a pure function.
    return response.startStream.sceneViewId.hex;
  }

  private async reconnectStreamingClient(streamId: string): Promise<void> {
    await this.connectStream(
      this.commands.execute<Disposable>('stream.reconnect', { sceneId: scene.id })
    );

    await this.stream.reconnect({ streamId, ...this.dimensions });
  }
```

Reply:
I think the implementations are different enough, and in one use case we want a new stream, where in another instance we do not (the initial start up), especially after the other pr feedback has been addressed. PTAL

Reply:
On second thought, I have made a common function for just the execution of the connection to the stream